shared/util/api.ts

```diff
@@ -42,28 +42,61 @@ export interface LineMatch {
     offsetAndLengths: [number, number][]
 }
 
+/** Metadata about a resolved repository. */
+export interface RepoMeta {
+    name: string
+    isFork: boolean
+    isArchived: boolean
+}
+
 export class API {
     /**
-     * Retrieves the name of a repository. Throws an error if the repository
-     * is not known to the Sourcegraph instance.
+     * Retrieves the name and fork/archive status of a repository. This method
+     * throws an error if the repository is not known to the Sourcegraph instance.
      *
      * @param cloneURL The repository's clone URL.
      */
-    public async resolveRepo(cloneURL: string): Promise<string> {
+    public async resolveRepo(cloneURL: string): Promise<RepoMeta> {
+        const metaFields = (await this.hasForkField())
+            ? 'isFork\nisArchived'
+            : ''
+
         const query = gql`
             query ResolveRepo($cloneURL: String!) {
                 repository(cloneURL: $cloneURL) {
                     name
+                    ${metaFields}
                 }
             }
         `
 
         interface Response {
-            repository: { name: string }
+            repository: RepoMeta
         }
 
         const data = await queryGraphQL<Response>(query, { cloneURL })
-        return data.repository.name
+        return data.repository
+    }
+
+    /** Determines via introspection if the GraphQL API has iSFork field on the Repository type. */
```

Comment:
Is this needed so that we can safely update the client before we update the GraphQL server implementation?
If so, then I assume that means we can get rid of this eventually—in which case could you please add a TODO to that effect somewhere nearby?

Reply:
I'm not sure what our contract for old instances are. We should decide that as well, but I'll add a comment here just in case we can eventually remove it.